core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,277 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for solution.
+ */
+
+oppia.controller('StateSolution', [
+  '$scope', '$rootScope', '$modal', '$filter', '$injector',
+  'editorContextService', 'alertsService', 'INTERACTION_SPECS',
+  'stateSolutionService', 'explorationStatesService',
+  'StateSolutionHelperService', 'oppiaExplorationHtmlFormatterService',
+  'stateInteractionIdService', 'stateHintsService', 'UrlInterpolationService',
+  'SolutionObjectFactory', 'responsesService', 'AnswerClassificationService',
+  'explorationContextService', 'angularNameService',
+  function(
+    $scope, $rootScope, $modal, $filter, $injector,
+    editorContextService, alertsService, INTERACTION_SPECS,
+    stateSolutionService, explorationStatesService,
+    StateSolutionHelperService, oppiaExplorationHtmlFormatterService,
+    stateInteractionIdService, stateHintsService, UrlInterpolationService,
+    SolutionObjectFactory, responsesService, AnswerClassificationService,
+    explorationContextService, angularNameService) {
+    $scope.editorContextService = editorContextService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.isActive = false;
+    $scope.answerCheckPassed = false;
+    $scope.INTERACTION_SPECS = INTERACTION_SPECS;
+    $scope.interactionHtml = '';
+
+    $scope.stateHintsService = stateHintsService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.correctAnswer = null;
+
+    $scope.submitAnswer = function (answer) {
+      $scope.correctAnswer = answer;
+    };
+
+    $scope.toggleIsActive = function() {
+      $scope.isActive = !$scope.isActive;
+    };
+
+    $scope.$on('stateEditorInitialized', function(evt, stateData) {
+      stateSolutionService.init(
+        editorContextService.getActiveStateName(),
+        stateData.interaction.solution);
+      $scope.interactionHtml = (
+        oppiaExplorationHtmlFormatterService.getInteractionHtml(
+          stateInteractionIdService.savedMemento,
+          explorationStatesService.getInteractionCustomizationArgsMemento(
+            editorContextService.getActiveStateName()),
+          'currentInteractionHtml'));
```

Comment:
Is this focus target unique to the solution editor? Should it have the word "solution" in it somewhere? Otherwise focus may go to the wrong place, leading to hard-to-debug errors.